samza-core/src/main/java/org/apache/samza/execution/JobPlanner.java

```diff
@@ -155,4 +158,33 @@ final void writePlanJsonFile(String planJson) {
         systemStreamConfigs.put(JobConfig.JOB_DEFAULT_SYSTEM(), dsd.getSystemName()));
     return systemStreamConfigs;
   }
+
+  /**
+   * Generates job.id from app.id and job.name from app.name config
+   * If both job.id and app.id is defined, app.id takes precedence and job.id is set to value of app.id
+   * If both job.name and app.name is defined, app.name takes precedence and job.name is set to value of app.name
+   *
+   * @param allowedUserConfig configs passed from user
+   */
+  void generateJobIdAndName(Map<String, String> allowedUserConfig) {
+    if (allowedUserConfig.containsKey(JobConfig.JOB_ID())) {
+      LOG.warn("{} is a deprecated configuration, use app.id instead.", JobConfig.JOB_ID());
```

Comment:
1. Can you please deprecate the job.id, job.name configurations from the samza configuration table. 
2. There're some samples in [samza-hello-samza](https://github.com/apache/samza-hello-samza/blob/master/src/main/config/azure-application-local-runner.properties) in which job.id, job.name are defined as configuration. Could you please update them to use app.id, app.name. Could you check if the same point applies to the samza-hello-samza samples within linkedin? 
3. Could you please update the integration test files located in samza-test/src/test/ to use the new app.id, app.name configuration. [Example](https://github.com/apache/samza/blob/master/samza-test/src/main/config/join/emitter.samza)
4. From what I see, we're doing configuration aliasing in the `LocalApplicationRunner`, `JobPlanner` directly. Could you share what is the rationale behind this decision? Why not do this in a `Rewriter`?
5. Consider a case where user has defined only `app.id`, `app.name` in the configuration and launches their low-level application. Could you please share how will submitting `YarnJob` through `JobRunner` work? `run-job.sh`  launches `YarnJob` which requires the  `job.name`, `job.id` to built through `CoordinatorStreamUtil.buildCoordinatorStreamConfig` ?

Reply:
Have you planned to migrate the existing tools like `CheckpointTool`, `CoordinatorStreamWriter` etc to use the same configuration aliasing ?

Reply:
Thanks for the review @shanthoosh 

1. Sure will do

2. We will be able to migrate this once a release containing these changes is published but I have that in mind

3. I changed a few, once again this change is user-facing for not using job.id and job.name. Hence we need to only change user-facing samples/prototypes/tests

4. Rewriters are a pluggable component, that is defined by a user. Changes I am proposing is independent of rewriting/aliasing. Changes I have proposed will make Samza framework generate job.id and job.name consistently and internally use it.  

5. I was planning to do those in another RB, but the changes are small enough so I included them here. 

Thanks